### DISCUSSION TITLE
For calculation of Jonswap/Pierson-Moskowitz spectrum

### DISCUSSION THREAD

**Discussion by icnosvbovwb** [OpenFAST/openfast#1397] [Q&A]
Dear @jjonkman 
![image](https://user-images.githubusercontent.com/107483730/210050934-5b9e40bb-1fad-4b4a-b0f8-674e402da9aa.png)
This figure shows that I set WaveMod=2, WaveHs=5, WaveTp=10, WavePkShp=1.835 in the hydrodyn.dat. Other settings are default. I used Jason_ PSD. m calculates the Wave1Elv PSD, but it is not as smooth as the curve shown in the next picture. I don't know why?
But waves need wind, that is, U10 mentioned below. How does it know the wind speed（U10) in openfast? I wonder if Hs and Tm are set, then openfast calculates U10 by itself and gets the corresponding wave spectrum.

![image](https://user-images.githubusercontent.com/107483730/210051723-04c26a8c-e336-4c60-ae40-8efefa70921f.png)
![image](https://user-images.githubusercontent.com/107483730/210051766-32e1376e-28c6-474e-b51a-6f0bd11c1822.png)
These two pictures show that I used the PlotBothSpectra. m script (https://github.com/amarnathajay/SeaSpectra) to calculate the spectrum of Jonswap and Pierson-Moskowitz spectrum.

Q1: I don't quite understand the relationship between Wave1Elv PSD calculated by the Jason_ PSD.m script and the spectrum calculated by the PlotBothSpectra. m script.

**Comment by jjonkman** [comment]
Dear @icnosvbovwb,

I'm not familiar with the _PlotBothSpectra.m_ script, which doesn't appear to be developed by NREL, so, I won't comment on that.

The spectrum plotted via _Jason_PSD.m_ is not smooth either because (1) you have generated the sea state with `WaveNDAmp` enabled in HydroDyn, which randomizes both the amplitude and phase) or (2) you have resampled the wave data in some way (the time series uses a time step and/or simulation length that is different from `WaveDT` and `WaveTMax`).

HydroDyn makes use of a Pierson-Woskowitz and/or JONSWAP spectrum that depends on HydroDyn inputs `WaveHs` and `WaveTp`, but not on wind speed.  (Wind speed in OpenFAST is defined within the InflowWind module, but this data is not used by HydroDyn).

Best regards,

**Comment by icnosvbovwb** [comment]
Dear @jjonkman 
![image](https://user-images.githubusercontent.com/107483730/210500377-c2cda9ce-c2ff-4696-a88f-9417c8468b7c.png)
![image](https://user-images.githubusercontent.com/107483730/210500427-f2020bdb-74dc-4c66-ba86-4d0b8ef0b0b4.png)
![image](https://user-images.githubusercontent.com/107483730/210499509-b91ebf7a-b413-4fe2-8878-52a40a5d65b3.png)

I modified WaveNDAmp=FALSE and set WavePkShp =1.
Data=xlsread('Wave1Elv.xlsx');
df=1/1000;  
N=1000/0.0125;
[f1,Sf1]=Jason_PSD(Data,df,N);
plot(f1,Sf1)
The spectrum plotted via Jason_PSD.m is not smooth，I don't know what's wrong.
